src/v/kafka/client/producer.cc

```diff
@@ -55,6 +54,10 @@ make_produce_response(model::partition_id p_id, std::exception_ptr ex) {
     } catch (const ss::gate_closed_exception&) {
         vlog(kclog.debug, "gate_closed_exception");
         response.error_code = error_code::operation_not_attempted;
+    } catch (const ss::sleep_aborted&) {
+        /// Could only occur when abort_source is triggered via stop()
+        vlog(kclog.debug, "sleep_aborted exception");
```

Comment:
I don't see a handler for the `abort_requested_exception` you're now returning from produce
```suggestion
    } catch (const ss::abort_requested_exception&) {
        /// Could only occur when abort_source is triggered via stop()
        vlog(kclog.debug, "abort_requested_exception");
```